### PR TITLE
[ENHANCEMENT] Docstrings for DataConnector and associated classes

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- DocStrings for Classes and public methods under `great_expectations/datasource/data_connectors/`
- ToDo : `data_connector/assets/asset.py`, `data_connector/sorters/*.py` and `data_connector/partition_query.py`

### CODE REVIEW

File: great_expectations/datasource/data_connector/configured_asset_filesystem_data_connector.py [hunk 1]
```diff
@@ -16,6 +16,17 @@
 
 
 class ConfiguredAssetFilesystemDataConnector(ConfiguredAssetFilePathDataConnector):
+    """
+    Extension of ConfiguredAssetFilePathDataConnector used to connect to Filesystem
+
+    The ConfiguredAssetFilesystemDataConnector is one of two classes (InferredAssetFilesystemDataConnector being the
+    other one) designed for connecting to filesystem-like data, more specifically files on disk. It connects to assets
```
Comment (alexsherstinsky) [inline great_expectations/datasource/data_connector/configured_asset_filesystem_data_connector.py]: In this case not filesystem-like but specifically filesystem data.

File: great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py [hunk 2]
```diff
@@ -22,6 +19,16 @@
 
 
 class ConfiguredAssetSqlDataConnector(DataConnector):
+    """
+    A DataConnector that requires explicit listing of Sql tables you want to connect to.
```
Comment (alexsherstinsky) [inline great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py]: Perhaps we should use SQL (all capitals in the acronym) when as part of documentation?  Thanks.

File: great_expectations/datasource/data_connector/inferred_asset_file_path_data_connector.py [hunk 3]
```diff
@@ -10,8 +10,17 @@
...
 
 class InferredAssetFilePathDataConnector(FilePathDataConnector):
-    """InferredAssetFilePathDataConnector is a base class for DataConnectors that operate on file paths and determine
-    the data_asset_name implicitly (e.g., through the combination of the regular expressions pattern and group names.
+    """
+    The InferredAssetFilePathDataConnector is one of two classes (ConfiguredAssetFilePathDataConnector being the
+    other one) designed for connecting to filesystem-like data. This includes files on disk, but also things
+    like S3 object stores, etc:
+
+    InferredAssetFilePathDataConnector is a base class that operates on file paths and determine
```
Comment (alexsherstinsky) [inline great_expectations/datasource/data_connector/inferred_asset_file_path_data_connector.py]: determines (in place of "determine")

File: great_expectations/datasource/data_connector/inferred_asset_filesystem_data_connector.py [hunk 4]
```diff
@@ -15,6 +15,18 @@
 
 
 class InferredAssetFilesystemDataConnector(InferredAssetFilePathDataConnector):
+    """
+    Extension of InferredAssetFilePathDataConnector used to connect to filesystem
+
+    The InferredAssetFilesystemDataConnector is one of two classes (ConfiguredAssetFilesystemDataConnector being the
+    other one) designed for connecting to filesystem-like data, more specifically files on disk. It connects to assets
```
Comment (alexsherstinsky) [inline great_expectations/datasource/data_connector/inferred_asset_filesystem_data_connector.py]: In this case not just filesystem-like but actually filesystem (analogous comment as for ConfiguredAssetFilesystemDataConnector).

**Review by alexsherstinsky — COMMENTED** [review]
@Shinnnyshinshin Please follow up on the comments as you see fit, and I will be happy to re-review; also, any docstring for RuntimeDataConnector planned?  Thank you so much.

File: great_expectations/datasource/data_connector/configured_asset_filesystem_data_connector.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/datasource/data_connector/configured_asset_filesystem_data_connector.py]: thank you for catching this

File: great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py (hunk 2 above)
Comment (Shinnnyshinshin) [reply great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py]: done

**Comment by Shinnnyshinshin** [comment]
@alexsherstinsky. If you're ok with it I'd like to address RuntimeDataConnector docstrings in a separate PR... I created a corresponding task in Asana. Could you let me know what you think?

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**